### ISSUE TITLE
Unable to chrome cast gen1 from HA without significant (30s) lag

### ISSUE THREAD

**Issue by atv2016** [AlexxIT/WebRTC#381]
I tried webrtc dash cast, with RTSP urls, restreamed url's from frigate, camera entity with RTSP, camera entities with restream RTSP, if it does show it's with a 30second lag.

Is there any documented way of using webrtc from HA to chromecast where there is nog lag ?

**Comment by AlexxIT** [comment]
Have you tried this urls in some desktop app? Ex. VLC

**Comment by atv2016** [comment]
Yes, works fine in VLC

**Comment by atv2016** [comment]
I am using webrtc v3 v4. The card is not really into play here, I don’t think I can use that to cast from an automation to chromecast?

I use webrtc dashcast or the camera integration.


________________________________

**Comment by AlexxIT** [comment]
If you using `webrtc.dash_cast` service - you using WebRTC Card on your Chromecast device.
Try use a direct RTSP link to the camera. Without using Frigate restream.

**Comment by atv2016** [comment]
I think i tried that, but i'll try again.
________________________________

**Comment by atv2016** [comment]
Good point about the bit stream, i'll try and use the sub stream to cast.

**Comment by atv2016** [comment]
So, i bought a chromecast HD, just to rule out any issues. For camera.stream, although it's now only 15seconds, rather then 30 seconds, it's still not nowhere realtime or 1-2 secs of lag. For WebRTC, it just hangs when i tries to load a RTSP url.I tried all codecs on the CCTV and all possible options i can think off on HA.

I don't really understand this. I'm going to check if my RTSP streams in XMeye (camera viewing tool) and frigate are realtime or if they are also lagging.

**Comment by AlexxIT** [comment]
Well you can compare with any old version at any time

**Comment by atv2016** [comment]
But how come old webrtc at least casted the rtsp streams, and the new one just sits there. The stream is fine in frigate or VLC. It’s the latest chromecast HD.

So it doesn’t look like the rtsp functionality is working in dashcast?
________________________________

**Comment by AlexxIT** [comment]
I have tested many RTSP cameras with Chromecast. All of them has 0.5 second delay.

**Comment by atv2016** [comment]
Yes but that’s not the problem. The problem is webrtc is not showing the stream at all.I would be happy with just 0.5 delay.
________________________________

**Comment by AlexxIT** [comment]
I don't understand you. You said you have stream but 30 seconds delay

**Comment by atv2016** [comment]
Sorry, yes but that was with camera entities from frigate or generic ones. With newer chromecast it’s actually about 15s delay.

But If I cast a full rtsp stream it just sits there. That’s what I meant. It doesn’t load at all. I tried enabling audio on cctv side or changing quality in the hopes of the codec changing but no go. It works fine in frigate or VLC. But I can ask VLC what codec it runs when it streams.

Thanks for your help.
________________________________

**Comment by AlexxIT** [comment]
I don't understand your again. Does any camera from any source work for you on Chromecast?

**Comment by atv2016** [comment]
Yes like I said, camera entities created from frigate or from generic entities created, work fine if I choose the camera entity option rather then the rtsp option.

It’s just when I use the rtsp option in webcast that it doesn’t work.
________________________________

**Comment by AlexxIT** [comment]
Well. What about delay on Chromecast for working cameras/entities? It should be 0.5 second. It can't be 15-30 second.

**Comment by atv2016** [comment]
Well ..it’s not :-). What can I say.
I heard this was normal for camera entities as they use HA/Hls but I don’t know.

If you have a way of resolving that I’m all ears.
________________________________

**Comment by AlexxIT** [comment]
Show how to you run Chromecast from HA

**Comment by atv2016** [comment]
I’ll post some screenshots how I use it
________________________________

**Comment by AlexxIT** [comment]
H264 means nothing. Because it can be with different profile and different levels.
What about "Show how to you run Chromecast from HA"?

**Comment by atv2016** [comment]
![Screenshot 2023-02-02 at 11 00 56](https://user-images.githubusercontent.com/16917203/216307444-598e6d08-c2de-4582-9f90-f3feb1f248e9.png)
![Screenshot 2023-02-02 at 10 59 09](https://user-images.githubusercontent.com/16917203/216307471-d11f7c5b-c66b-40ff-9abf-bca1b2bc66a8.png)
![Screenshot 2023-02-02 at 10 59 53](https://user-images.githubusercontent.com/16917203/216307483-96fae500-47a7-4255-a62e-65d44730867f.png)

**Comment by atv2016** [comment]
I will add a screenshot of how webrtc is just hanging there, it opens it then it shows a black screen with a bar at the bottom but no video. I tried restream RTSP urls, real RTSP, entities, generic entities etc.
